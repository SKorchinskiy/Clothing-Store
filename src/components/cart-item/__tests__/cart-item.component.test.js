import { screen, render } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import "@testing-library/jest-dom";

import CartItem from "../cart-item.component";

import {
  cartItem,
  increaseItemQuantity,
  decreaseItemQuantity,
  clearItemFromCart,
} from "../../../stubs/cart-item.stub";

describe("<CartItem /> component", () => {
  it("should render cart-item UI elements", () => {
    const { asFragment } = render(<CartItem cartItem={cartItem} />);

    expect.assertions(1);
    expect(asFragment()).toMatchSnapshot();
  });

  it("should call provided method when button is clicked", () => {
    render(
      <CartItem
        cartItem={cartItem}
        increaseItemQuantity={increaseItemQuantity}
        decreaseItemQuantity={decreaseItemQuantity}
        clearItemFromCart={clearItemFromCart}
      />
    );

    expect.assertions(3);

    const increaseQuantityElement = screen.getByText("❭");
    userEvent.click(increaseQuantityElement);
    expect(increaseItemQuantity).toHaveBeenCalledWith(cartItem);

    const decreaseQuantityElement = screen.getByText("❬");
    userEvent.click(decreaseQuantityElement);
    expect(decreaseItemQuantity).toHaveBeenCalledWith(cartItem);

    const clearItemElement = screen.getByText("✕");
    userEvent.click(clearItemElement);
    expect(clearItemFromCart).toHaveBeenCalled();
  });
});
