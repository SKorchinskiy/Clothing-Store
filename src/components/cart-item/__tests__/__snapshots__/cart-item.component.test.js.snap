// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CartItem /> component should render cart-item UI elements 1`] = `
<DocumentFragment>
  <div
    class="sc-dkjKgF jouCEI"
  >
    <img
      alt="item 1"
      src="testUrl"
    />
    <div
      class="sc-bcPKhP qrmOf"
    >
      <div
        class="sc-grXZZQ bkNFIY"
      >
        item 1
      </div>
      <div
        class="sc-grXZZQ bkNFIY"
      >
        <div
          class="arrow"
        >
          ❬
        </div>
        10
        <div
          class="arrow"
        >
          ❭
        </div>
      </div>
      <div
        class="sc-grXZZQ bkNFIY"
      >
        $140
      </div>
      <div
        class="sc-grXZZQ bkNFIY"
      >
        <div
          class="remove-btn"
        >
          ✕
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
